### DISCUSSION TITLE
What is the easiest way to automatically download and update software?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [dotnet/runtime#102419] [Q&A]
I have written a C# program, which does not use any installation program. As I am a beginner and do not know how to use an installer, I simply used AOT to publish the program in Visual Studio, then zipped it and handed it over to the customer.

The program periodically checks whether the version number on my PHP-written website is higher than its own version number. If it is, then it will notify the customer to download the new zip package from the website. 

However, I think this is not good enough since other software can update automatically with just one click. 

So, how can I achieve this in C# using the simplest method possible, like a NuGet package? I am a beginner, so complex technologies are not suitable for me at this stage.

**Comment by huoyaoyuan** [comment]
To be simple, you can find for some integrated solutions like [Squirrel](https://github.com/Squirrel), [ClickOnce](https://learn.microsoft.com/visualstudio/deployment/quickstart-deploy-using-clickonce-folder), or get helped by app store for Windows/macOS. There's no standard answer for this.
